drivers/SmartThings/matter-switch/src/eve-energy/init.lua

```diff
@@ -0,0 +1,231 @@
+local cluster_base = require "st.matter.cluster_base"
+local utils = require "st.utils"
+
+local EVE_MANUFACTURER_ID = 0x130A
+local PRIVATE_CLUSTER_ID = 0x130AFC01
+
+local PRIVATE_ATTR_ID_WATT = 0x130A000A
+local PRIVATE_ATTR_ID_WATT_ACCUMULATED = 0x130A000B
+
+local LAST_REPORT_TIME = "LAST_REPORT_TIME"
+local TIMER_REPEAT = (1 * 60)    -- Run the timer each minute
+local REPORT_TIMEOUT = (10 * 60) -- Report the value each 10 minutes
+
+
+-------------------------------------------------------------------------------------
+-- Eve specifics
+-------------------------------------------------------------------------------------
+
+local function is_eve_energy_products(opts, driver, device)
+	if device.manufacturer_info.vendor_id == EVE_MANUFACTURER_ID then
+		return true
+	end
+
+	return false
+end
+
+-- Return a ISO 8061 formatted timestamp in UTC (Z)
+-- @return e.g. 2022-02-02T08:00:00Z
+local function iso8061Timestamp(time)
+	return os.date("!%Y-%m-%dT%TZ", time)
+end
+
+local function updateEnergyMeter(device, totalConsumptionWh)
+	-- Report the energy consumed
+	device:emit_event(capabilities.energyMeter.energy({ value = totalConsumptionWh, unit = "Wh" }))
+
+	-- Only send powerConsumptionReport every 10 minutes
+	local current_time = os.time()
+	local last_time = device:get_field(LAST_REPORT_TIME) or 0
+	local next_time = last_time + REPORT_TIMEOUT
+	if current_time < next_time then
+		return
+	end
+
+	device:set_field(LAST_REPORT_TIME, current_time, { persist = true })
+
+	-- Calculate the energy consumed between the start and the end time
+	local previousTotalConsumptionWh = device:get_latest_state("main", capabilities.powerConsumptionReport.ID,
+		capabilities.powerConsumptionReport.powerConsumption.NAME)
+
+	local deltaEnergyWh = 0.0
+	if previousTotalConsumptionWh ~= nil and previousTotalConsumptionWh.energy ~= nil then
+		deltaEnergyWh = math.max(totalConsumptionWh - previousTotalConsumptionWh.energy, 0.0)
+	end
+
+	local startTime = iso8061Timestamp(last_time)
+	local endTime = iso8061Timestamp(current_time - 1)
+
+	-- Report the energy consumed during the time interval. The unit of these values should be 'Wh'
+	device:emit_event(capabilities.powerConsumptionReport.powerConsumption({
+		start = startTime,
+		["end"] = endTime,
+		deltaEnergy = deltaEnergyWh,
+		energy = totalConsumptionWh
+	}))
+end
+
+
+-------------------------------------------------------------------------------------
+-- Timer
+-------------------------------------------------------------------------------------
+
+local function requestData(device)
+	-- Update the on/off status
+	device:send(clusters.OnOff.attributes.OnOff:read(device))
+
+	-- Update the Watt usage
+	device:send(cluster_base.read(device, 0x01, PRIVATE_CLUSTER_ID, PRIVATE_ATTR_ID_WATT, nil))
+
+	-- Update the energy consumption
+	device:send(cluster_base.read(device, 0x01, PRIVATE_CLUSTER_ID, PRIVATE_ATTR_ID_WATT_ACCUMULATED, nil))
+end
+
+local timer = nil
```

Comment:
note that this is a driver-level variable and this will have unintended results if more than one device is using this driver

you'll want to use the device storage for this (`device:set_field`)

Reply:
Because of some unintended changesets in this PR, I created a new PR at #940

In the new PR, the timer has been changed from a driver-level variable to a device storage